connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -1465,6 +1465,49 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     }
   }
 
+  test("test unsigned integer types") {
+    // The java type for uint32 and uint64 is signed integer and long respectively.
+    // Let's check that we're converting correctly
+    val sample = spark.range(1).select(
+      lit(
+        UnsignedTypes
+          .newBuilder()
+          .setUint32Value(Integer.MIN_VALUE)
+          .setUint64Value(Long.MinValue)
+          .build()
+          .toByteArray
+      ).as("raw_proto"))
+
+    val expected = spark.range(1).select(
+      struct(
+        lit(Integer.toUnsignedLong(Integer.MIN_VALUE).longValue).as("uint32_value"),
+        lit(BigDecimal(java.lang.Long.toUnsignedString(Long.MinValue))).as("uint64_value")
+      ).as("proto")
+    )
+
+    val expectedWithLegacy = spark.range(1).select(
+      struct(
+        lit(Integer.MIN_VALUE).as("uint32_value"),
```

Comment:
using `MIN_VALUE` as its all `1`s in binary and is the largest possible number if its "unsigned"